app/src/main/java/ch/epfl/sweng/SDP/Matchmaking/User.java

```diff
@@ -0,0 +1,181 @@
+package ch.epfl.sweng.SDP.Matchmaking;
+
+import android.annotation.TargetApi;
+import android.os.Build;
+
+import android.util.Log;
+
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import static java.lang.Math.toIntExact;
+
+public class User implements Serializable {
```

Comment:
We already have a class called Account which implements most of these functions. Andrea is creating a locally cached instance of the user account where you can retrieve the stars and trophies from. Talk to him to also store matches played/won.